docs/src/stake-pool.md

```diff
@@ -1,11 +1,427 @@
+```
+
+In return, the stake pool has sent us staking derivatives in the form of SPL
+tokens.  We can double-check our stake pool account using the SPL token
+command-line utility.
+
+```sh
+$ spl-token balance 34XMHa3JUPv46ftU4dGHvemZ9oKVjnciRePYMcX3rjEF
+0.024058966
+```
+
+### Example: Update
+
+Every epoch, the network pays out rewards to stake accounts managed by the stake
+pool, increasing the value of staking derivative SPL tokens minted on deposit.
+In order to calculate the proper value of these stake pool tokens, we must update
+the total value managed by the stake pool every epoch.
+
+The Solana transaction processor has two important limitations:
+
+* size of the overall transaction, limited to roughly 1 MTU / packet
+* computation budget per instruction
+
+A stake pool may manage hundreds of staking accounts, so it is impossible to
+update the total value of the stake pool in one instruction. Thankfully, the
+command-line utility does all of the work of breaking up transactions.
+
+```sh
+$ spl-stake-pool update --pool 3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC 
+Signature: 3Yx1RH3Afqj5ckX8YvPCRt1DudVP4HuRPkh1dBPvTM9GqGxcB9ZXHGZPADVSZiaqKi166fevMG232EWxrRWswPtt
+```
+
+If another user already updated the stake pool balance for the current epoch, we
+see a different output.
+
+```sh
+$ spl-stake-pool update --pool 3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC 
+Stake pool balances are up to date, no update required.
+```
+
+### Example: Withdraw stake
+
+Whenever we want to recover SOL plus accrued rewards, we can provide our
+staking derivative SPL tokens in exchange for an activated stake account.
+
+Let's withdraw 0.02 of our staking derivative tokens from the stake pool.
+
+```sh
+$ spl-stake-pool withdraw --pool 3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC  --amount 0.02 --burn-from 34XMHa3JUPv46ftU4dGHvemZ9oKVjnciRePYMcX3rjEF
+Withdrawing from account FYQB64aEzSmECvnG8RVvdAXBxRnzrLvcA3R22aGH2hUN, amount 8.867176821 SOL, 0.019999999 pool tokens
+Creating account to receive stake CZF2z3JJoDmJRcVjtsrz1BKUUGNL3VPW5FPFqge1bzmQ
+Signature: 2xBPVPJ749AE4hHNCNYdjuHv1EdMvxm9uvvraWfTA7Urrvecwh9w64URCyLLroLQ2RKDGE2QELM2ZHd8qRkjavJM
+```
+
+Our 0.02 pool tokens were taken, and in exchange we received a fully active stake
+account, delegated to `2HUKQz7W2nXZSwrdX5RkfS2rLU4j1QZLjdGCHcoUKFh3`.  We can
+double-check the status of the stake account:
+
+```sh
+$ solana stake-account CZF2z3JJoDmJRcVjtsrz1BKUUGNL3VPW5FPFqge1bzmQ
+Balance: 8.869459257 SOL
+Rent Exempt Reserve: 0.00228288 SOL
+Delegated Stake: 8.867176377 SOL
+Active Stake: 8.867176377 SOL
+Delegated Vote Account Address: 2HUKQz7W2nXZSwrdX5RkfS2rLU4j1QZLjdGCHcoUKFh3
+Stake Authority: 4SnSuUtJGKvk2GYpBwmEsWG53zTurVM8yXGsoiZQyMJn
+Withdraw Authority: 4SnSuUtJGKvk2GYpBwmEsWG53zTurVM8yXGsoiZQyMJn
+```
+
+Note: due to truncation on SPL token to SOL conversion, we will not receive the
+exact amount calculated by the command-line utility.
```

Comment:
Can the cli be changed to show the exact amount received?   This seems like an annoyance the user shouldn't need to have to deal with, and could breed mistrust

Reply:
I did some refactoring and managed to fix this :-)